lib/ansible/modules/packaging/language/pip.py

```diff
@@ -235,6 +241,73 @@
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
 
+class Distribution:
+    def __init__(self, name_string, version_string=None):
+        if not name_string.startswith(('svn+', 'git+', 'hg+', 'bzr+', 'file:')):
+            self._plain_distribution = True
+            if version_string:
+                version_string = version_string.lstrip()
+                separator = '==' if version_string[0].isdigit() else ' '
+                name_string = separator.join((name_string, version_string))
+            self._requirement = Requirement.parse(name_string)
+            self._distribution_name = self._requirement.project_name
+        else:
+            self._plain_distribution = False
+            self._distribution_name = name_string
+
+    @property
+    def has_version_specifier(self):
+        if self._plain_distribution:
+            return bool(self._requirement.specs)
+        return False
+
+    @property
+    def distribution_name(self):
+        return self._distribution_name
+
+    @distribution_name.setter
+    def distribution_name(self, new_name):
+        self._distribution_name = new_name
+
+    def is_satisfied_by(self, version_to_test, module):
+        if self._plain_distribution and hasattr(self._requirement, 'specifier'):
+            return self._requirement.specifier.contains(version_to_test)
+        else:
+            module.warn(str(self._requirement.__dict__))
+        return False
+
+    def __str__(self):
+        if self._plain_distribution:
+            return str(self._requirement)
+        return self._distribution_name
+
+
+def _is_valid_distribution_name(name):
+    name = name.lstrip()
+    return not name.startswith(('>=', '<=', '!=', '==', '>', '<'))
+
+
+def _recover_distribution_name(names):
+    distribution_parts = []
+    for name in names:
+        if _is_valid_distribution_name(name):
+            if len(distribution_parts) > 0:
+                yield ",".join(distribution_parts)
```

Comment:
I'd probably add `distribution_parts = []` after yield inside of `if` block and then you'd only need `distribution_parts.append(name)` outside of if-block without a need to have diverse code or `else`-block.